modules/apps/portal-search/portal-search-rest-test/src/testIntegration/java/com/liferay/portal/search/rest/resource/v1_0/test/SearchResultResourceTest.java

```diff
@@ -446,6 +478,25 @@ private void _testPostSearchPageWithDateRangeFacetConfiguration()
 		Assert.assertEquals("1", termJSONObject.getString("displayName"));
 	}
 
+	private void _testPostSearchPageWithEmbeddedNestedFields(
+			ObjectDefinition objectDefinition)
+		throws Exception {
+
+		String entryClassName = objectDefinition.getClassName();
+
+		SearchPage<SearchResult> page = _postSearchPage(
+			entryClassName, null, objectDefinition.getUserName(), "embedded",
+			new SearchRequestBody());
+
+		Collection<SearchResult> items = page.getItems();
+
+		Assert.assertFalse(items.isEmpty());
```

Comment:
I don't think this assert is necessary because it's implicitly "tested" on line  495

Reply:
if there is no an `Assert.assertFalse(items.isEmpty())` in this line and `items` is empty, the code inside "for" will never be executed and so the test will not fail. That's why this check is important, as the test will fail if `items` is empty.

Reply:
Right :)